tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -5121,9 +5121,11 @@ class HLSLExternalSource : public ExternalSemaSource {
       const TemplateArgument &arg = argLoc.getArgument();
       if (arg.getKind() == TemplateArgument::ArgKind::Type) {
         QualType argType = arg.getAsType();
-        if (!IsValidTemplateArgumentType(argSrcLoc, argType, requireScalar)) {
-          // NOTE: IsValidTemplateArgumentType emits its own diagnostics
-          return true;
+        if (!argType->isDependentType()) {
```

Comment:
A short comment explaining that these will be checked when they become concrete might be helpful